samza-core/src/main/scala/org/apache/samza/metrics/reporter/MetricsSnapshotReporterFactory.scala

```diff
@@ -77,15 +71,48 @@ class MetricsSnapshotReporterFactory extends MetricsReporterFactory with Logging
     } else {
       new MetricsSnapshotSerdeV2
     }
-
     info("Got serde %s." format serde)
+    serde
+  }
+
 
-    val pollingInterval: Int = metricsConfig.getMetricsSnapshotReporterInterval(name)
+  def getBlacklist(reporterName: String, config: Config): Option[String] = {
+    val metricsConfig = new MetricsConfig(config)
+    val blacklist = JavaOptionals.toRichOptional(metricsConfig.getMetricsSnapshotReporterBlacklist(reporterName)).toOption
+    info("Got blacklist as: %s" format blacklist)
+    blacklist
+  }
 
-    info("Setting polling interval to %d" format pollingInterval)
+  def getPollingInterval(reporterName: String, config: Config): Int = {
+    val metricsConfig = new MetricsConfig(config)
+    val pollingInterval = metricsConfig.getMetricsSnapshotReporterInterval(reporterName)
```

Comment:
why is this `pollingInterval` instead of `reportingInterval`? makes me think we poll for metrics at this interval but then the error statement above which says "Error reporting ... `pollingInterval`" is confusing.

Reply:
Good point! I'm not sure why this was called `pollingInterval` to begin with.
Fixed it.